tools/validation-generator/src/main/java/io/spine/tools/validate/MessageValidatorFactory.java

```diff
@@ -0,0 +1,237 @@
+                violation -> formatted("violations.add(%s)", violation);
+        Expression msg = Expression.of(MESSAGE_PARAMETER);
+        FieldValidatorFactories factories = new FieldValidatorFactories(msg);
+        for (FieldDeclaration field : type.fields()) {
+            FieldValidatorFactory factory = factories.forField(field);
+            Optional<CodeBlock> fieldValidation = factory.generate(violationAccumulator);
+            fieldValidation.ifPresent(body::add);
+        }
+        body.addStatement(RETURN_N, VIOLATIONS);
+        return body.build();
+    }
+
+    /**
+     * Generates the {@code validate()} method for the message class.
+     *
+     * <p>The method is {@code public} and returns a list of {@link ConstraintViolation}s if
+     * the message is not valid. For the full contract, see
+     * {@link io.spine.protobuf.MessageWithConstraints#validate()}.
+     *
+     * @return {@code validate()} method
+     */
+    public MethodSpec generateValidate() {
+        CodeBlock body = CodeBlock.of("return $T.$N(this);",
+                                      bestGuess(validatorSimpleName), VALIDATE_METHOD);
+        return MethodSpec
+                .methodBuilder(VALIDATE_METHOD)
+                .addModifiers(PUBLIC)
+                .addAnnotation(Beta.class)
+                .addAnnotation(Override.class)
+                .returns(listOfViolations)
+                .addCode(body)
+                .build();
+    }
+
+    /**
+     * Generates the {@code vBuild()} method for the message builder class.
+     *
+     * @return {@code vBuild()} method
+     * @see io.spine.protobuf.ValidatingBuilder#vBuild() for the full contract.
+     */
+    public MethodSpec generateVBuild() {
+        ClassName messageClass = bestGuess(messageSimpleName.value());
+        Class<ValidationException> exceptionClass = ValidationException.class;
+        @SuppressWarnings("DuplicateStringLiteralInspection")
+        CodeBlock body = CodeBlock
+                .builder()
+                .addStatement("$T $N = build()",
+                              messageClass, MESSAGE_VARIABLE)
+                .addStatement("$T $N = $N.$N()",
+                              listOfViolations,
+                              VIOLATIONS,
+                              MESSAGE_VARIABLE,
+                              VALIDATE_METHOD)
+                .beginControlFlow("if (!$N.isEmpty())", VIOLATIONS)
+                .addStatement("throw new $T($N)", exceptionClass, VIOLATIONS)
+                .endControlFlow()
+                .addStatement(RETURN_N, MESSAGE_VARIABLE)
+                .build();
+        return MethodSpec
+                .methodBuilder("vBuild")
+                .addAnnotation(Override.class)
+                .addModifiers(PUBLIC)
+                .returns(messageClass)
+                .addException(exceptionClass)
+                .addCode(body)
+                .build();
```

Comment:
I wonder why not to use `vBuild()` in the messages like this one...

Reply:
`MethodSpec` is not a message, just a Java API of JavaPoet.
If this doesn't answer your question, let's discuss.